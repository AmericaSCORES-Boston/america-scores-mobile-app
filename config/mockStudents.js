export const student1 = {
  'id': 1,
  'name': 'John Doe',
  'dob': '03-29-2002',
  'height': 15,
  'weight': null
};

export const student2 = {
  'id': 2,
  'name': 'Jeremy Parker',
  'dob': '08-12-1996',
  'height': 10,
  'weight': 50
};

export const student3 = {
  'id': 3,
  'name': 'Sally Smith',
  'dob': '02-05-1997',
  'height': null,
  'weight': null
};

export const badStudent = {
  'id': 4,
  'name': 'Grim Reaper',
  'dob': '01-01-1900',
  'height': 'tall',
  'weight': null,
  'living': false
};

export const students = [
  student1,
  student2,
  student3
];

export const studentsWithHeightAndWeight = [
  student2
];

export const studentsWithoutHeightAndWeight = [
  student1,
  student3
];
