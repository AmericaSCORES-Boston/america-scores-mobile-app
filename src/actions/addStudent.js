
import { ADD_STUDENT, ADD_STUDENT_SUCCESS, ADD_STUDENT_FAILURE } from '../actions/constants';

export function addStudent(student) {
  return {
    type: ADD_STUDENT,
    student,
  };
}

export function addStudentSuccess() {
  return {
    type: ADD_STUDENT_SUCCESS,
  };
}

export function addStudentFailure() {
  return {
    type: ADD_STUDENT_FAILURE,
  };
}
