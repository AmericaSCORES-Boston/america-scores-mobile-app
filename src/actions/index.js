export {
  fetchSites,
  fetchSitesSuccess,
  fetchSitesFailure
} from './site';

export {
  fetchPrograms,
  fetchProgramsSuccess,
  fetchProgramsFailure,
  addProgram,
  addProgramSuccess,
  addProgramFailure
} from './program';

export {
  fetchStudents,
  fetchStudentsSuccess,
  fetchStudentsFailure
} from './student';

export {
    fetchStat,
    fetchStatSuccess,
    createStat,
    createStatSuccess,
    updateStat,
    updateStatSuccess,
    statRequestFailure
} from './studentStat';

export {
    fetchStats,
    fetchStatsSuccess,
    fetchStatsFailure
} from './stat';
