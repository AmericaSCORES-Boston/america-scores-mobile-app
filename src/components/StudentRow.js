import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 12,
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    marginLeft: 12,
    fontSize: 16,
  },
});

const StudentRow = (student) => (
  <TouchableOpacity onPress={() => student.switchToDetailRoute(student)}>
    <View style={styles.container}>
      <Text>{ student.name }</Text>
      <Text style={styles.text}>{ student.number}</Text>
    </View>
  </TouchableOpacity>
);

export default StudentRow;
