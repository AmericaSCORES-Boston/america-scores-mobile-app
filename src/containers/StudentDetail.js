import { connect } from 'react-redux'
import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

class StudentDetail extends React.Component {
  render() {
    return (
      <Text>Hi</Text>
    );
  }
}

export default StudentDetail;
