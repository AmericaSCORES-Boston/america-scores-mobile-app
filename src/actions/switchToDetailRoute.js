import { SWITCH_TO_STUDENT_DETAIL_ROUTE } from '../actions/constants';

export default (student) => ({
  type: SWITCH_TO_STUDENT_DETAIL_ROUTE,
  student,
});
