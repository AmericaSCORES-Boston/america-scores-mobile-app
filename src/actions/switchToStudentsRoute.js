import { SWITCH_TO_STUDENTS_ROUTE } from '../actions/constants';

export default () => ({
  type: SWITCH_TO_STUDENTS_ROUTE,
});
