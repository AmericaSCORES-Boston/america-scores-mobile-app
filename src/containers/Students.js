import React, { Component, PropTypes } from 'react';
import { Actions } from 'react-native-router-flux';
import { View, Text } from 'react-native';
import { Container, Content, List, ListItem, Footer, FooterTab, Button, H2, H3 } from 'native-base';
import scoresTheme from '../themes/scoresTheme';
import { connect } from 'react-redux';
import * as studentActions from '../actions/student';
import * as eventActions from '../actions/event';
import dates from '../util/dates';

import styles from '../styles';

class StudentsContainer extends Component {
  constructor(props) {
    super(props);
    const student_ids = [],
        eventToday = null;
    this.state = {...props.studentsState, ...props.eventsState, student_ids, eventToday};

    this.props.component.onRight = () => {
      Actions.addStudent({program_id: this.props.program_id});
    };

    this.props.component.rightTitle = 'Add';
  }

  componentWillMount() {
    this.props.fetchStudents(this.props.program_id);
    this.props.fetchEvents(this.props.program_id);
  }

  componentWillReceiveProps(nextProps) {
    const newStudentsState = nextProps.studentsState,
        newEventsState = nextProps.eventsState;
    if (newStudentsState && newStudentsState.students) {
      this.state = newStudentsState;
    }

    if (newEventsState.events && this.state.events !== newEventsState.events) {
      this.state.events = newEventsState.events;
      const filteredEvents = this.state.events.slice().filter(function(event) {
        return dates.getDateStringFromSql(event.event_date) === dates.getTodayDateString();
      });
      if (filteredEvents.length > 0) {
        this.setState({eventToday: filteredEvents[0]});
      }
    }
  }

  // Does this program have any students?
  hasStudents() {
    const students = this.state.students || [];
    return students.length > 0;
  }

  render() {
    return (
        <Container style={styles.container} theme={scoresTheme}>
          <Content>
            <View>
              {this.showStudents()}
            </View>
          </Content>

          {this.showFooter()}

        </Container>
    );
  }


  // Show a list of students for this program if it is not empty.
  // Otherwise, show a message about the program being empty.
  showStudents() {
    const students = (
        <List
            dataArray={this.state.students}
            renderRow={(rowData) =>
                <ListItem button onPress={()=>Actions.individualStudent({title: rowData.first_name + ' ' + rowData.last_name, student: rowData})}>
                  <Text>{rowData.first_name + ' ' + rowData.last_name}</Text>
                </ListItem>
              }
        />
    );

    const noStudents = (
        <View>
          <H3 style={[styles.textAlignCenter, styles.mediumVerticalMargin]}>There are no students registered for this program.</H3>
          <H2 style={styles.textAlignCenter}>Click the 'Add' button to get started.</H2>
        </View>
    );

    return (this.hasStudents()) ? students : noStudents;
  }


  // Show a footer with a pacer and bmi collection button if this program has students.
  // Otherwise, hide the footer.
  showFooter() {
    const program = {program_name: this.props.title, program_id: this.props.program_id};
    const footer = (
        <Footer>
          <FooterTab>
            <Button active onPress={()=>Actions.pacer({students: this.state.students, event: this.state.eventToday})}>
              Pacer Test
            </Button>

            <Button active onPress={()=>Actions.bmi({program, students: this.state.students, event: this.state.eventToday})}>
              BMI Collection
            </Button>

          </FooterTab>
        </Footer>
    );

    return (this.hasStudents()) ? footer : null;
  }
}

const mapStateToProps = (state) => ({
  studentsState: state.studentsState,
  eventsState: state.eventsState
});

const mapDispatchToProps = (dispatch) => ({
  fetchStudents: (program_id) => {
    dispatch(studentActions.fetchStudents(program_id));
  },
  fetchEvents: (program_id) => {
    dispatch(eventActions.fetchEvents(program_id));
  }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(StudentsContainer);
