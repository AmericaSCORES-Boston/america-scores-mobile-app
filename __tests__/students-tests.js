import React from 'react';
import renderer from 'react-test-renderer';
import Students from '../src/containers/Students';
import StudentRow from '../src/components/StudentRow';
import StudentDetail from '../src/containers/StudentDetail';
import * as MockStudents from '../config/mockStudents';
/**
 * *************************************************
 *   COMPONENT TESTS
 * *************************************************
 */
it('renders the Students page', () => {
    const tree = renderer.create(
        <Students students={MockStudents} />
).toJSON();
expect(tree).toMatchSnapshot();
});

it('renders StudentDetail page', () => {
    const tree = renderer.create(
        <StudentDetail student={MockStudents.student3} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders StudentRow button correctly', () => {
    const tree = renderer.create(
        <StudentRow student={MockStudents.student1} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});


/**
 * *************************************************
 *   ACTION CREATOR TESTS
 * *************************************************
 */
import * as addStudent from '../src/actions/addStudent';
import * as fetchStudents from '../src/actions/fetchActions';
import * as switchToDetail from '../src/actions/switchToDetailRoute';
import * as switchStudents from '../src/actions/switchToStudentsRoute';

it('creates a "ADD_STUDENT" action', () => {
    expect(addStudent.addStudent(MockStudents.student1)).toMatchSnapshot();
});

it('creates an "ADD_STUDENT_SUCCESS" action', () => {
    expect(addStudent.addStudentSuccess()).toMatchSnapshot();
});

/**
//  * *************************************************
//  *   REDUCER TESTS
//  * *************************************************
//  */
// import * as reducer from '../src/reducers/';
//
// it('returns the same state on an unhandled action', () => {
//     expect(reducer({type: '_NULL'})).toMatchSnapshot();
// });
//
// it('starts BMI collection with a filtered list', () => {
//     const action = actions.startBMICollection(MockStudents.students);
//     expect(reducer(action)).toMatchSnapshot();
// });
