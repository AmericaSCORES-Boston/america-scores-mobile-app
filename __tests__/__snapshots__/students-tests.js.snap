exports[`test renders StudentDetail page 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  Hi
</Text>
`;

exports[`test renders StudentRow component correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "padding": 12,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail" />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "marginLeft": 12,
        }
      } />
  </View>
</View>
`;

exports[`test renders the Students page 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": NaN,
    }
  }
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderRow={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flex": 1,
      "marginTop": 20,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "padding": 12,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        John Doe
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": 12,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "padding": 12,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Jeremy Parker
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": 12,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "padding": 12,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Sally Smith
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": 12,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "padding": 12,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Grim Reaper
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": 12,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "padding": 12,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail" />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": 12,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "padding": 12,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail" />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": 12,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "padding": 12,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail" />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": 12,
          }
        } />
    </View>
  </View>
</ScrollView>
`;
